### PR TITLE
add corner resize, add rotate

### PR DESCRIPTION
**Pull request by Zonciu**
`rect` `path` `polygon` shapes tested

### CODE REVIEW

File: src/svg.resize.js [hunk 1]
```diff
@@ -51,22 +50,36 @@ class ResizeHandler {
...
-    // We consider the resize done, when a touch is canceled, too
-    const eventMove = (isMouse ? 'mousemove' : 'touchmove') + '.resize'
-    const eventEnd = (isMouse ? 'mouseup' : 'touchcancel.resize touchend') + '.resize'
+    if (e.type === 'rot') { // rotate
+      if (this.el.dispatch('beforerotate', { event: e, handler: this }).defaultPrevented) {
+        return
+      }
+      // We consider the resize done, when a touch is canceled, too
+      const eventMove = (isMouse ? 'mousemove' : 'touchmove') + '.rotate'
+      const eventEnd = (isMouse ? 'mouseup' : 'touchcancel.rotate touchend') + '.rotate'
```
Comment (Fuzzyma) [inline src/svg.resize.js]: There is no reason for replicating this lines here. The only thing different for rotation is the function which is called as event handler. Everything else is teh same.

So do the eventMove = and eventEnd = part once before everything. You also dont need an extra event dispatch for rotation. The event is passed along in details. The user can see what it was

File: src/svg.resize.js [hunk 2]
```diff
@@ -113,11 +126,34 @@ class ResizeHandler {
...
+    let angle = Math.acos((dx1 * dx2 + dy1 * dy2) / c) / Math.PI * 180
+    if (endPoint.x < this.startPoint.x) {
+      angle = -angle
+    }
+    if (this.el.dispatch('rotate', { angle: angle, event: e, handler: this }).defaultPrevented) {
+      return
+    }
+    this.el.rotate(angle)
+  }
+  endRotate (ev) {
```
Comment (Fuzzyma) [inline src/svg.resize.js]: endRotate and endResize is teh same when you made the changes from the other comment. So you dont need an extra method here. Simplification is key! :)

File: src/svg.resize.js [hunk 3]
```diff
@@ -113,11 +126,34 @@ class ResizeHandler {
...
+    const c = Math.sqrt(dx1 * dx1 + dy1 * dy1) * Math.sqrt(dx2 * dx2 + dy2 * dy2)
+    if (c === 0) return 0
+    let angle = Math.acos((dx1 * dx2 + dy1 * dy2) / c) / Math.PI * 180
+    if (endPoint.x < this.startPoint.x) {
+      angle = -angle
+    }
+    if (this.el.dispatch('rotate', { angle: angle, event: e, handler: this }).defaultPrevented) {
+      return
+    }
+    this.el.rotate(angle)
```
Comment (Fuzzyma) [inline src/svg.resize.js]: el.rotate will do a relative rotation. I am mildly confused that this worked for you. It should spin like crazy - strange

File: src/svg.resize.js [hunk 4]
```diff
@@ -113,11 +126,34 @@ class ResizeHandler {
 
   endResize (ev) {
     this.resize(ev)
```
Comment (Zonciu) [inline src/svg.resize.js]: What's this resize call use for?

File: src/svg.resize.js [hunk 5]
```diff
@@ -51,21 +49,26 @@ class ResizeHandler {
       return
     }
 
+    this.box = this.el.bbox()
+    this.startPoint = this.el.point(startX, startY)
```
Comment (Fuzzyma) [inline src/svg.resize.js]: you dont need to run this functions when the event gets preventen in the next line. So move them under the event check to potentially save time

File: src/svg.resize.js [hunk 6]
```diff
@@ -51,21 +49,26 @@ class ResizeHandler {
...
+
     // Fire beforedrag event
     if (this.el.dispatch('beforeresize', { event: e, handler: this }).defaultPrevented) {
```
Comment (Fuzzyma) [inline src/svg.resize.js]: We need to pass the event here so people know what they are preventing

File: src/svg.resize.js [hunk 7]
```diff
@@ -111,10 +114,42 @@ class ResizeHandler {
     this.el.move(box.x, box.y).size(box.width, box.height)
   }
 
+  rotate (e) {
+    const endPoint = this.el.point(getCoordsFromEvent(e))
+    const cx = this.el.cx()
+    const cy = this.el.cy()
```
Comment (Fuzzyma) [inline src/svg.resize.js]: you already have this knowledge saved in this.box.cx and this.box.cy. You dont want to call bbox TWICE on every mousemove (cx and cy both call bbox)

File: src/svg.resize.js (hunk 3 above)
Comment (Fuzzyma) [reply src/svg.resize.js]: I still have no idea why this works :D

File: src/svg.resize.js [hunk 8]
```diff
@@ -81,40 +84,74 @@ class ResizeHandler {
...
-    off(window, 'mousemove.resize touchmove.resize', this.resize)
+    // Unbind resize and end events to window
+    if (this.eventType === 'rot') { // rotate
+      off(window, 'mousemove.resize touchmove.resize', this.rotate)
+    } else if (this.eventType === 'shear') { // shear
+      off(window, 'mousemove.resize touchmove.resize', this.shear)
+    } else { // resize
+      this.resize(ev)
+      off(window, 'mousemove.resize touchmove.resize', this.resize)
+    }
```
Comment (Fuzzyma) [inline src/svg.resize.js]: Alright we can do better. Just use `off(window, 'mousemove.resize touchmove.resize')` for everything!

**Comment by Fuzzyma** [comment]
Nice work!

**Pull request merged**